tests/test_github_requirements.py

```diff
@@ -0,0 +1,113 @@
+        return self._test_require_status('org/project1',
+                                         'project1-pipeline')
+
+#    def test_trigger_require_status(self):
+#        "Test trigger requirement: status"
+#        return self._test_require_status('org/project2',
+#                                         'project2-trigger')
+
+    def _test_require_status(self, project, job):
+        self.config.set('zuul', 'layout_config',
+                        'tests/fixtures/layout-github-requirement-status.yaml')
+        self.sched.reconfigure(self.config)
+        self.registerJobs()
+
+        A = self.fake_github.openFakePullRequest(project, 'master', 'A')
+        # A comment event that we will keep submitting to trigger
+        comment = A.getCommentAddedEvent('test me')
+        self.fake_github.emitEvent(comment)
+        self.waitUntilSettled()
+        # No status from zuul so should not be enqueued
+        self.assertEqual(len(self.history), 0)
+
+        # An error status should not cause it to be enqueued
+        A.setStatus('zuul/check', 'error')
+        self.fake_github.emitEvent(comment)
+        self.waitUntilSettled()
+        self.assertEqual(len(self.history), 0)
+
+        # A success status goes in
+        A.setStatus('zuul/check', 'success')
+        self.fake_github.emitEvent(comment)
+        self.waitUntilSettled()
+        self.assertEqual(len(self.history), 1)
+        self.assertEqual(self.history[0].name, job)
+
+    def test_pipeline_require_reject_status(self):
+        "Test negative pipeline requirement: status with state"
+        return self._test_require_reject_status('org/project1',
+                                                'project1-pipeline')
+
+#    def test_trigger_require_reject_status_state(self):
+#        "Test negative trigger requirement: status with state"
+#        return self._test_require_reject_status('org/project2',
+#                                                'project2-trigger')
+
+    def _test_require_reject_status(self, project, job):
+        "Test negative status match"
+        # Should only trigger if zuul hasn't set error status
+        self.config.set(
+            'zuul', 'layout_config',
+            'tests/fixtures/layout-github-requirement-reject-status.yaml')
+        self.sched.reconfigure(self.config)
+        self.registerJobs()
+
+        # add in a change with no comments
+        A = self.fake_github.openFakePullRequest(project, 'master', 'A')
+        self.waitUntilSettled()
+        self.assertEqual(len(self.history), 0)
+
+        # add in a comment that will trigger
+        comment = A.getCommentAddedEvent('test me')
+        self.fake_github.emitEvent(comment)
+        self.waitUntilSettled()
+        self.assertEqual(len(self.history), 1)
+        self.assertEqual(self.history[0].name, job)
+
+        # add in a status from which shouldn't trigger
+        A.setStatus('zuul/check', 'error')
+        self.fake_github.emitEvent(comment)
```

Comment:
is the purposely emitting a comment event here? or should setting the status be emitting its own event? as it is, we're re-using the comment event from line 91 which is actually triggering the thing here.